14_contains/solution/contains-solution.js

```diff
@@ -0,0 +1,16 @@
+const contains = function(obj, searchValue) {
+  const values = Object.values(obj);
+
+  // NaN !== NaN, so we would usually have to do an explicit check with Math.isNaN to test for it. However, Array.prototype.includes handles that for us
+  if (values.includes(searchValue)) return true;
+
+  const nestedObjects = values.filter((value) => typeof value === 'object' && value !== null);
+  for (const nestedObject of nestedObjects) {
+    return contains(nestedObject, searchValue);
+  }
+
+  return false;
```

Comment:
Not even just a nit - I just realised that specifically due to the test object not testing for sibling objects (since it's all just children), it all passes despite us returning on the first iteration of the for..of loop.

The for..of approach would need to be
```js
for (const nestedObject of nestedObjects) {
  if (contains(nestedObject, searchValue)) {
    return true;
  }
}
```
which the `Array.prototype.some` approach solves just fine when accounting for values present in non-first nested objects.

I think `.some` is appropriate for where people would be in this part of the course in JS :+1:

Reply:
I have slightly re-written the solution to use a queue now so this suggestion is no longer applicable to the new approach (with a queue)